x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ApiKeyAuthenticatorTests.java

```diff
@@ -72,4 +84,199 @@ public void testAuditingOnAuthenticationTermination() {
         }
     }
 
+    public void testRecordingSuccessfulAuthenticationMetrics() {
```

Comment:
nit: Some of this test code is duplicated, might benefit from moving some of this in to methods.

Reply:
Agreed. I'll iterate on it and dry it up a bit.